File: gen_token.py
import hmac, hashlib, json, base64, time

secret = 'arizen-dev-secret-change-in-production'
header = base64.urlsafe_b64encode(json.dumps({"alg":"HS256","typ":"JWT"}).encode()).rstrip(b'=').decode()
payload = base64.urlsafe_b64encode(json.dumps({
    "name": "Test Student",
    "email": "test@example.com",
    "sub": "test-user-id",
    "role": "LEARNER",
    "iat": int(time.time()),
    "exp": int(time.time()) + 3600,
}).encode()).rstrip(b'=').decode()
sig = hmac.new(secret.encode(), f"{header}.{payload}".encode(), hashlib.sha256).digest()
sig_b64 = base64.urlsafe_b64encode(sig).rstrip(b'=').decode()
token = f"{header}.{payload}.{sig_b64}"
with open('.jwt_token', 'w') as f:
    f.write(token)
print("Token written to .jwt_token")
print("TOKEN:", token)

File: gen_token.js
// Generate a JWT token for dev auth bypass
// The session endpoint (/api/auth/session) decodes without verifying signature
const header = Buffer.from(JSON.stringify({alg:'HS256','typ':'JWT'})).toString('base64url');
const now = Math.floor(Date.now() / 1000);
const payload = Buffer.from(JSON.stringify({
  sub: 'test-user-id',
  email: 'test@example.com',
  name: 'Test Student',
  role: 'LEARNER',
  guildId: null,
  guildSlug: null,
  learnerProfileId: null,
  grade: 4,
  displayName: 'Test Student',
  totalXp: 0,
  currentStreak: 0,
  avatarUrl: null,
  iat: now,
  exp: now + 3600 * 24 * 30,
})).toString('base64url');
const sig = Buffer.from('fake-sig-for-dev').toString('base64url');
const token = header + '.' + payload + '.' + sig;
console.log(token);
